### DISCUSSION TITLE
esp_core_dump_flash: Core dump flash config is corrupted! CRC=0x7bd5c66f instead of 0x0

### DISCUSSION THREAD

**Discussion by TIMESTICKING** [HomeSpan/HomeSpan#490] [Q&A]
I got completely no clue.... well this happened on my second board, it's esp32-S, and I google it up, it's actually NODEMCU-32s, but the core of it is still esp32 with 2 CPU cores.

My first board is ESP-WROOM-32, it works fine, but when I change it to my second board, it failed, it reboots forever, and printed
```
ELF file SHA256: ddfd500a0a2a4f08

E (442) esp_core_dump_flash: Core dump flash config is corrupted! CRC=0x7bd5c66f instead of 0x0
Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13104
load:0x40080400,len:3036
entry 0x400805e4
E (417) cpu_start: Running on single core variant of a chip, but app is built with multi-core support.
E (417) cpu_start: Check that CONFIG_FREERTOS_UNICORE is enabled in menuconfig

abort() was called at PC 0x40082f0d on core 0


Backtrace: 0x40083959:0x3ffe3b80 0x4008d419:0x3ffe3ba0 0x400930a1:0x3ffe3bc0 0x40082f0d:0x3ffe3c40 0x400792ba:0x3ffe3c90 |<-CORRUPTED
```
and for the error, I looked it up, and someone says: 

 **You're running code on a single-core chip (the board you're using has an ESP32-S0WD chip, which is a single-core variant of the dual-core ESP32), but the code you're trying to run was compiled for a dual-core CPU (most likely an ESP32).
As far as I could find, the ESP32-S0WD is not supported by the ESP32 Arduino Core**

so I guess homespan doesn't support this board? Or is there any code that I could edit to avoid this issue?

**Comment by TIMESTICKING** [comment]
well, It seems like its the problem of arduino, which is compiled on dual-core esp32... I'll be damned...

**Reply by HomeSpan** [reply]
Did you set the right chip type in the Arduino IDE?  NODEMCU-32S is one of the choices.  You can always try compiling a simple test-sketch (not a HomeSpan sketch) to make sure you selected the correct board.

**Reply by TIMESTICKING** [reply]
I tried a lot of boards, but none of them worked. What actually surprised me is that even using choose nodemcu-32s in PIO, it still shows `Running on single core variant of a chip, but app is built with multi-core support.`, I thought they would realize it and compile the Arduino framework of nodemcu with one single core.
Anyway, I gave up on running Arduino on one core board, I just ordered 4 esp32-wroom boards LOL...

**Comment by HBSuperfreak** [comment]
> I got completely no clue.... well this happened on my second board, it's esp32-S, and I google it up, it's actually NODEMCU-32s, but the core of it is still esp32 with 2 CPU cores.
> 
> My first board is ESP-WROOM-32, it works fine, but when I change it to my second board, it failed, it reboots forever, and printed
> 
> ```
> ELF file SHA256: ddfd500a0a2a4f08
> 
> E (442) esp_core_dump_flash: Core dump flash config is corrupted! CRC=0x7bd5c66f instead of 0x0
> Rebooting...
> ets Jun  8 2016 00:22:57
> 
> rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
> configsip: 0, SPIWP:0xee
> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
> mode:DIO, clock div:2
> load:0x3fff0030,len:1184
> load:0x40078000,len:13104
> load:0x40080400,len:3036
> entry 0x400805e4
> E (417) cpu_start: Running on single core variant of a chip, but app is built with multi-core support.
> E (417) cpu_start: Check that CONFIG_FREERTOS_UNICORE is enabled in menuconfig
> 
> abort() was called at PC 0x40082f0d on core 0
> 
> 
> Backtrace: 0x40083959:0x3ffe3b80 0x4008d419:0x3ffe3ba0 0x400930a1:0x3ffe3bc0 0x40082f0d:0x3ffe3c40 0x400792ba:0x3ffe3c90 |<-CORRUPTED
> ```

H. I just got a new board and right on first connnect Im getting the same error codes.  Ctrl v'd below. I actually went and flashed the firmware just in case that's what was needed. Same result. Altho I did get the flash done properly going by the espressif.com page.
Uploaded a simple sketch (example from R. Tillard. ) same result. So far I've found very little on this error. Have you or anyone found the cause? Kind of want to avoid the hassle of returns... Thanks!

(Erro code, repeats every few ms;
E (150) esp_core_dump_flash: Core dump flash config is corrupted! CRC=0x7bd5c66f instead of 0x0
Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13924
ho 0 tail 12 room 4
load:0x40080400,len:3600
entry 0x400805f0
assert failed: do_core_init startup.c:326 (flash_ret == ESP_OK)

**Reply by TIMESTICKING** [reply]
Hi, the problem is solved because I changed a new chip, which is an actual esp32 wroom. The issue occurred is due to the chip you use is not dual core, Arduino should be running on a two chip esp. So... I guess you had to purchase a new one

**Reply by saepien** [reply]
I have an ESP32 DevKitC with ESP WROOM 32D which is a dual-core chip. Having exactly this issue.

**Reply by fenilGhoghari** [reply]
> 

Hi, 
Actually you didn't need to change your hardware every controller have they own boot-loader and Arduino IDE provide you to select proper boot-loader you can re-select your appropriate boot-loader and check it will work not an issue with Hardware maybe....

**Reply by Fanjio01** [reply]
Hi,
In case it helps anyone: I solved this issue by selecting DIO as flash mode instead of QIO.
![ESP32_CAM Arduino Config](https://github.com/HomeSpan/HomeSpan/assets/126648491/0e194a23-7068-4e68-87d1-24cb385d0637)

Regards

**Reply by zhang-yuan-dong** [reply]
Thanks for sharing!!

**Reply by jsieberg** [reply]
DIO solved my problem.
Thanks

**Comment by mhamidjamil** [comment]
i have solved my issue @Fanjio01 thanks your solution solve my problem, I was getting this error before apply your solution:
ELF file SHA256: c70c06fd85d0c3de

E (181) esp_core_dump_flash: Core dump flash config is corrupted! CRC=0x7bd5c66f instead of 0x0
Rebooting...
⸮

**Reply by Fanjio01** [reply]
Hi, i'm glad it helped :)

**Reply by Stewie-040882** [reply]
DIO flash fixed for me too! :-D 
Thanks for Sharing

**Comment by teddy1908** [comment]
Hi all,

I ran into a similar problem and am completely lost now.

Namely I have 2 self deisgned boards which worked till now and 2 development boards which also worked. Today I tried to load a new firmware version to one of my boards. When I loaded the new firmware everything collapsed, same happened to the second one and even when trying to revert to an older version of the firmware or just loading a simple arduino sample I am still getting these messages 

![image](https://github.com/HomeSpan/HomeSpan/assets/153741006/ea51f658-9fbc-4852-af74-318ac992e9cc)

I tried using a web flasher to erase the flash of the ESP 32 S3 I am using but I cannot solve this. Its stuck in this  loop of errors.

Also ... here are my Arduino programming setings

![image](https://github.com/HomeSpan/HomeSpan/assets/153741006/af43d325-921a-4b02-bc73-fb7abb61efbc)

Does anybody have an Idea what I did?

regards
Tadej

**Reply by rodcoles** [reply]
I had this issue with a esp32-s3-wroom-2 and i fixed it by changing the flash mode to "OPI 80mhz"

**Reply by teddy1908** [reply]
Thanks ... I changed to OPI and it works now. 
The thing is ... I do not know why it only works in OPI mode. As far as I understand this is the slowest. I currently do not need faster operation but what settings do I need to have to have the ESP working in DIO?

**Reply by Hit1009** [reply]
@teddy1908 @rodcoles 
can you please guide me how to change the flash mode to OPI 80mhz? as I don't see any OPI option in the "Tools" menu
![image](https://github.com/HomeSpan/HomeSpan/assets/116697144/6a685977-917d-4676-9eb5-e7efe1e0b09b)

**Reply by Hit1009** [reply]
@teddy1908 @rodcoles  this is what I see on the serial monitor

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1184
load:0x40078000,len:13260
load:0x40080400,len:3028
entry 0x400805e4
E (128) spi_flash: Detected size(4096k) smaller than the size in the binary image header(8192k). Probe failed.

assert failed: do_core_init startup.c:328 (flash_ret == ESP_OK)


Backtrace: 0x40083349:0x3ffe3ab0 0x40087f45:0x3ffe3ad0 0x4008ce29:0x3ffe3af0 0x400d9096:0x3ffe3c20 0x40082a75:0x3ffe3c50 0x40079306:0x3ffe3c90 |<-CORRUPTED




ELF file SHA256: 8f2b459c888d7075

E (151) esp_core_dump_flash: Core dump flash config is corrupted! CRC=0x7bd5c66f instead of 0x0
Rebooting...
ets Jul 29 2019 12:21:46

**Reply by arslan437** [reply]
I also got the same error like you. Did you found the solution ? 

```
E (220) esp_core_dump_flash: Core dump flash config is corrupted! CRC=0x7bd5c66f instead of 0x0
Rebooting...
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0xc (RTC_SW_CPU_RST),boot:0x2b (SPI_FAST_FLASH_BOOT)
Saved PC:0x40376c7c
  #0  0x40376c7c in esp_restart_noos at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/esp_system/port/soc/esp32s3/system_internal.c:151 (discriminator 1)

SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0x4bc
load:0x403c9700,len:0xbd8
load:0x403cc700,len:0x2a0c
entry 0x403c98d0
E (149) spi_flash: Detected size(4096k) smaller than the size in the binary image header(8192k). Probe failed.

assert failed: do_core_init startup.c:328 (flash_ret == ESP_OK)


Backtrace: 0x4037706e:0x3fceb180 0x4037a441:0x3fceb1a0 0x4037fca5:0x3fceb1c0 0x42017e66:0x3fceb2f0 0x403769df:0x3fceb320 0x403cd86b:0x3fceb350 0x403cdb2a:0x3fceb380 0x403c9925:0x3fceb4b0 0x40045c01:0x3fceb570 0x40043ab6:0x3fceb6f0 0x40034c45:0x3fceb710
  #0  0x4037706e in panic_abort at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/esp_system/panic.c:408
  #1  0x4037a441 in esp_system_abort at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/esp_system/esp_system.c:137
  #2  0x4037fca5 in __assert_func at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/newlib/assert.c:85
  #3  0x42017e66 in do_core_init at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/esp_system/startup.c:320
      (inlined by) start_cpu0_default at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/esp_system/startup.c:446
  #4  0x403769df in call_start_cpu0 at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/esp_system/port/cpu_start.c:687
```

**Reply by faizannazir** [reply]
i am not able to enter in download mode after first flash it keep resetting

**Reply by HomeSpan** [reply]
This is unrelated to HomeSpan.  

Try using one of the built-in non-HomeSpan examples and at the options to erase Flash before uploading.  

Also, make sure you have set the correct flash size, speed, type, etc for your board.  

The issue you are having usually means a download was interrupted, you have a corrupted partition table, etc.

**Reply by faizannazir** [reply]
yes it is not related to homespan it is related to esp32 i want to know from there guys who posted solution did they tested on 2nd pcb or on same pcb because i am not able to enter in download mode

**Comment by HomeSpan** [comment]
These types of issues do not seem related to HomeSpan, but rather how you are flashing the chip.  Recommend starting clean by trying to flash one of the basic ESP32 examples (e.g. Blink) but first set the Arduino IDE to ERASE the Flash before uploading the sketch.

**Reply by faizannazir** [reply]
i flashed empty sketch but i am not able to enter in download mode after first flash it keep resetting

**Comment by anindya19** [comment]
Please select the ESP32 Dev module and settings below to resolve this issue.
![image](https://github.com/user-attachments/assets/a090ea0c-b024-43f7-b7ab-79db6ec4d848)

**Reply by faizannazir** [reply]
but i am not able to enter in download mode after first flash esp is keep reseting

**Comment by sseruzaifa** [comment]
does any of you who solved it use vscode and platformio? what would the solution be in that case, because adding the DIO setting in platformio.ini does not seem to solve it for me. Thank you for all the submissions.
![image](https://github.com/user-attachments/assets/22d5e081-93f9-4b4e-8035-26bc8c442c4a)


@anindya19 @jsieberg @faizannazir

**Comment by faizannazir** [comment]
![IMG-20241006-WA0024](https://github.com/user-attachments/assets/092f37f9-42f4-4a86-b888-faf150e0511e)
@sseruzaifa run this esptool command and check

**Reply by sseruzaifa** [reply]
thank you for the support but I realized I couldnt use esptool commands since I am using the arduino environment. they only work when the project was created with esp-idf, unless there is something I needto learn

**Reply by faizannazir** [reply]
Arduino also use esptool to flash esp32  you can download it is too it command line tool

**Comment by ahentati** [comment]
Hello Chaps,

I'm getting a similar backtrace:


```
21:00:58.273 -> entry 0x400805cc
21:00:58.635 -> E (409) esp_timer: Task is already initialized
21:00:58.635 -> E (409) cpu_start: init function 0x400e867c has failed (0x103), aborting
21:00:58.667 -> 
21:00:58.667 -> abort() was called at PC 0x400e580f on core 0
21:00:58.667 -> 
21:00:58.667 -> 
21:00:58.667 -> Backtrace: 0x40082525:0x3ffe3b20 0x4008de29:0x3ffe3b40 0x40093129:0x3ffe3b60 0x400e580f:0x3ffe3be0 0x400e58d8:0x3ffe3c10 0x400828ae:0x3ffe3c40 0x400797e5:0x3ffe3c80 |<-CORRUPTED
21:00:58.667 -> 
21:00:58.667 -> 
21:00:58.667 -> 
21:00:58.667 -> 
21:00:58.667 -> ELF file SHA256: bd4b8f8853d76cf9
21:00:58.667 -> 
21:00:58.667 -> E (433) esp_core_dump_flash: Core dump flash config is corrupted! CRC=0x7bd5c66f instead of 0x0
21:00:58.699 -> E (441) esp_core_dump_elf: Elf write init failed!
21:00:58.699 -> E (446) esp_core_dump_common: Core dump write failed with error=-1
21:00:58.699 -> Rebooting...

```

Here is the board:

```
Chip is ESP32-D0WD-V3 (revision v3.1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
```

Any clue how I can fix this?

**Reply by HomeSpan** [reply]
Use one of the Espressif built-in examples, such as the simple Blink sketch (available from the Arduino IDE).  Under the Tools menu, set "Erase all flash before upload" to ENABLE.  Re-compile and upload again.

**Reply by ahentati** [reply]
Thx for the hint, but it didn't work. Uploading other sketches is working (probably smaller)... I tried to change the partition with no luck :(

**Comment by prayagkr** [comment]
Below configuration worked me.
I also removed dependency for DHT. DHT module was creating issue for me

![Screenshot 2025-02-24 at 22 55 13](https://github.com/user-attachments/assets/def4a445-b358-42e9-8bd0-4b2ca3a1d9f9)

**Reply by dwilliss** [reply]
I would like to point out another cause of this error.
I had a sketch that worked fine, but I made some changes and uploaded and suddenly got the "flash config is corrupted! CRC=0x7bd5c66f instead of 0x0" error (although with a different CRC value).
I uploaded the simple blink sketch and it was fine.
re-uploaded mine and error again.
I reverted my change and it worked.

It turns out that **any exception that happens in a constructor of an object declared outside of the setup or loop will cause the error**.
In my case, I had a constructor that tried to register its instance in a std::list that had not yet been constructed.
